ccw.core/src/clj/ccw/leiningen/classpath_container.clj

```diff
@@ -162,19 +170,22 @@
 
 (defn get-project-dependencies
   "Return the dependencies sorted alphabetically via their file name.
-   Throws Aether exceptions if a problem occured"
-  [project-name lein-project]
+   Throws Aether exceptions if a problem occurred. Supports thread
+  interruption. The function break!, a function that presumably throws
+  an exception), is called were appropriate to break execution."
+  [break! project-name lein-project]
   (let [dependencies (resolve-dependencies project-name :dependencies lein-project)
         default-native-platform-path (u/lein-native-platform-path lein-project)
         srcmap (get-source-map lein-project)]
     (t/format :leiningen "default-native-platform-path: %s" default-native-platform-path)
-    (->> dependencies
-      (filter #(re-find #"\.(jar|zip)$" (.getName ^File %)))
-      (sort-by #(.getName ^File %))
-      (map #(ser-dep %
-                     default-native-platform-path
-                     (get srcmap %)
-                     #_(or #_(u/lein-native-dependency-path lein-project %) ;; TODO make this work :-(
+    (for [dep (->> dependencies
+                   (filter #(re-find #"\.(jar|zip)$" (.getName ^File %)))
+                   (sort-by #(.getName ^File %)))]
+      (do (break!)
```

Comment:
`for` is not an imperative for-loop. It will return a lazy sequence. So you don't know when the call to `break!` will occur, unless you explicitly `(dorun)` over the sequence. but here, then, it would be more appropriate to directly use `doseq`, I guess.